File: stats.pl
#!/usr/bin/perl
open F, "</tmp/data.txt";

while (<F>) {
  $sum += $_;
  $n++;
}

$mean = $sum/$n;
seek(F, 0, 0);

$sum = 0;
while (<F>) {
  $sum += ($_ - $mean)**2;
}
printf "Mean: %0.2f\nVariance: %0.2f\n", $mean, $sum/$n;

File: stats.rb
sum=0
num=0
File.open("/tmp/data.txt") do |file|
  file.each do |line| 
    num=num+1 
    sum=sum+"#{line}".to_i
  end
end
puts "average is ",sum/num
aver=sum/num;
sum=0
File.open("/tmp/data.txt") do |file|
  file.each do |line|
    sum=sum+(("#{line}".to_i-aver)**2)
  end
end
puts "varience is ",sum/num
